toqito/matrix_ops/commutant.py

```diff
@@ -0,0 +1,104 @@
+"""Module for computing the commutant of a set of matrices."""
+
+import numpy as np
+from scipy.linalg import null_space
+
+
+def commutant(A: np.ndarray | list[np.ndarray]) -> list[np.ndarray]:
+    r"""Compute an orthonormal basis for the commutant algebra.
+
+    Given a set of matrices :math:`A`, this function determines an orthonormal basis
+    (with respect to the Hilbert-Schmidt inner product) for the algebra of matrices
+    that commute with every matrix in :math:`A`.
+
+    The commutant of a set of matrices consists of all matrices that commute
+    with every matrix in the set. Formally, given a matrix :math:`A \in \mathbb{C}^{n \times n}`,
+    a matrix :math:`X \in \mathbb{C}^{n \times n}` belongs to the commutant if:
+
+    .. math:: A X = X A \quad \forall A \in \mathcal{A}.
+
+    This can be rewritten as:
+
+    .. math::
+        (A \otimes I - I \otimes A^T) \text{vec}(X) = 0.
+
+    where :math:`\text{vec}(X)` denotes the vectorization of :math:`X`, which arranges its columns
+    into a single column vector. The null space of this equation provides a basis for the commutant
+    :cite:`QETLAB_link`.
+
+    Examples
+    ==========
+
+    Consider the following set of matrices:
+
+    .. math::
+        A_1 = \begin{pmatrix}
+                1 & 0 \\
+                0 & -1
+            \end{pmatrix}, \quad
+        A_2 = \begin{pmatrix}
+                0 & 1 \\
+                1 & 0
+            \end{pmatrix}
+
+    The commutant consists of matrices that commute with both :math:`A_1` and :math:`A_2`.
+
+    >>> import numpy as np
+    >>> from toqito.matrix_ops import commutant
+    >>>
+    >>> A1 = np.array([[1, 0], [0, -1]])
+    >>> A2 = np.array([[0, 1], [1, 0]])
+    >>> basis = commutant([A1, A2])
+    >>> basis
+    [array([[ 0.70710678,  0.        ],
+           [-0.        ,  0.70710678]])]
```

Comment:
```suggestion
    [array([[0.70710678, 0.        ],
           [0.        , 0.70710678]])]
```

Reply:
made this change to fix doctest failure

Reply:
Nice. We should definitely opt for aiming for the appropriate rendering of the output, and not use (ideally) any of the ELIPSIS, WHITESPACE, etc. doc modifiers as these are bandaid approaches to the actual problem.